src/economy/economy.cc

```diff
@@ -521,20 +521,21 @@ bool Economy::needs_ware_or_worker(DescriptionIndex const ware_or_worker_type) c
 	}  // Target quantity is set to 0, we need to check if there is an open request.
 	// For soldier requests, do not recruit new rookies if only heroes are needed.
 	const bool is_soldier = type_ == wwWORKER && ware_or_worker_type == owner().tribe().soldier();
-	for (const Request* req : requests_) {
-		if (req->get_type() == type_ && req->get_index() == ware_or_worker_type && req->is_open() &&
-		    (!is_soldier ||
-		     req->get_requirements().check(soldier_prototype(
-		        owner().egbase().descriptions().get_worker_descr(ware_or_worker_type))))) {
-			return true;
-		}
-	}
-	return false;
+	return std::any_of(
+	   requests_.begin(), requests_.end(),
+	   [this, is_soldier, ware_or_worker_type](const Request* req) {
+		   return req->get_type() == type_ && req->get_index() == ware_or_worker_type &&
+		          req->is_open() &&
+		          (!is_soldier ||
+		           req->get_requirements().check(soldier_prototype(
+		              owner().egbase().descriptions().get_worker_descr(ware_or_worker_type))));
+	   });
 }
 
 bool Economy::has_building(const DescriptionIndex di) const {
 	const std::string& name = owner().tribe().get_building_descr(di)->name();
-	for (const auto& bld : owner().get_building_statistics(di)) {
+	const auto& stats = owner().get_building_statistics(di);
+	return std::any_of(stats.begin(), stats.end(), [this, &name](const auto& bld) {
```

Comment:
I can't see any advantage from using any_o here over the old code. maybe anding th different if conditions might result in a proper predicate function